lib/std/os/windows/ws2_32.zig

```diff
@@ -116,7 +116,7 @@ pub const WSAOVERLAPPED_COMPLETION_ROUTINE = fn (dwError: DWORD, cbTransferred:
 pub const ADDRESS_FAMILY = u16;
 
 // Microsoft use the signed c_int for this, but it should never be negative
-const socklen_t = u32;
+pub const socklen_t = u31;
```

Comment:
this doesn't seem correct.

Reply:
Seems better than allowing `u32` while the Windows API expects a `i32`/`c_int`. There has to be a check somewhere that we don't go over `max(c_int)`. And if we don't I'd say we should just stick to the Windows api types and not change them.

Reply:
There are apis that take a `*socklen_t`: we can't pass a `*u31` to these.

Reply:
Yes, I encountered those. If we just want to pass pointers there, I'd suggest we just use `i32`.

Reply:
I'm ok with this changing to i32. But it should be u32 or i32 to match the ABI. The only question is where to put the checks.

Reply:
I've changed the `socklen_t` to `u32` and used `@intCast` and `@ptrCast` to convert to the Windows API which uses `c_int`. Since it's used only for the `sockaddr` struct it's unlikely that anyone puts huge numbers in there so I think this should suffice.